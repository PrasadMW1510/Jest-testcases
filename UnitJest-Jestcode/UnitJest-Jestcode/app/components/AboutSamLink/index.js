/**
 *
 * AboutSamLink
 *
 */

export { default } from './AboutSamLink';
