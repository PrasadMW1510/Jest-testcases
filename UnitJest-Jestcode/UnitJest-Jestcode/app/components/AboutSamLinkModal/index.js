/**
 *
 * MessageLogModal
 *
 */

export { default } from './AboutSamLinkModal';
