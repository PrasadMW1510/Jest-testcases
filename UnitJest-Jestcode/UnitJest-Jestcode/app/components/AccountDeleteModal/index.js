/**
 *
 * AccountDeleteModal
 *
 */

export { default } from './AccountDeleteModal';
