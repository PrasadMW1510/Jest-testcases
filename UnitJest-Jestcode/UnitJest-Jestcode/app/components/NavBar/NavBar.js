import React from 'react';
import PropTypes from 'prop-types';
import './NavBar.scss';

function NavBar({
  activeItemId,
  children: childrenProp,
  className,
  inset,
  palette: paletteProp,
  theme: themeProp,
  ...props
}) {
  // Setup classes
  const theme = themeProp && `navbar--theme-${themeProp}`;
  const palette = paletteProp && `navbar--palette-${paletteProp}`;
  // Set the appropriate nav item to `active`
  const children = React.Children.map(childrenProp, child => {
    if (!React.isValidElement(child)) {
      return null;
    }
    return React.cloneElement(child, {
      active: child.props.active || (child.props.id && child.props.id === activeItemId),
    });
  });
  return (
    <nav
      className={`navbar ${theme || ''} ${palette || ''} ${
        inset ? 'navbar--inset' : ''
      } ${className}`}
      {...props}
    >
      {children}
    </nav>
  );
}

NavBar.defaultProps = {
  className: '',
  inset: false,
};

NavBar.propTypes = {
  // Set the active NavItem by including the id of the NavItem.
  activeItemId: PropTypes.string,
  children: PropTypes.node.isRequired,
  // Use this to override any styles needed
  className: PropTypes.string,
  // Set to true for modals. It removes the left vertical line on first NavItem.
  inset: PropTypes.bool.isRequired,
  // The color scheme to use ('orange' || 'green')
  palette: PropTypes.string,
  // The theme to use (only 'tabs' currently)
  theme: PropTypes.string,
};

export default NavBar;
