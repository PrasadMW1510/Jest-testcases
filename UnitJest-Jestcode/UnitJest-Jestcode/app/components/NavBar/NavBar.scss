.navbar {
  position: relative;
  width: 100%;
}

.navbar__item {
  cursor: pointer;

  &:focus {
    outline: none;
  }

  &:hover,
  .navbar__item-text:hover {
    cursor: pointer;
  }
}

.navbar__item-text {
  color: #000;
  font-weight: 700;
  line-height: 26px; // Matches icon image height
  padding: 0 16px;
  position: relative;
  z-index: 2;

  &.navbar__item-text--error {
    color: #f00;
  }

}

// `inset` style removes left vertical line of first item -- style handly for modal use.
.navbar--inset .navbar__item {
  &:first-child::before {
    visibility: hidden; // `display: none` causes display issue here in couple browsers.
  }
}

//// Theme -> Tabs ////

.navbar--theme-tabs {
  border-bottom: 2px #000 solid;
  height: 32px;
}

.navbar--theme-tabs .navbar__item {
  align-items: center;
  border-style: none;
  box-sizing: border-box;
  display: inline-flex;
  height: 32px;
  overflow: hidden;
  position: relative;
  text-decoration: none;
  width: 170px;
  z-index: 10;
  // A background needed here to hide bottom border
  &.navbar__item--active {
    color: #fff;
    z-index: 12;
  }

  &.navbar__item--active::after {
    background-color: #fff;
  }

  // Active and last item are always in full view
  &.navbar__item--active,
  &:last-child {
    width: 180px;
  }

  &.navbar__item--active:last-child {
    margin-right: 10px;
  }

  // Create slanted borders
  &::before {
    background: #000;
    content: '';
    height: 32px;
    left: 0;
    position: absolute;
    top: 0;
    width: 2px;
    z-index: 9;
  }

  // Create item's vertical border on left-side
  &::after {
    border: 2px #000 solid;
    box-sizing: border-box;
    content: '';
    height: 32px;
    left: -16px;
    position: absolute;
    top: 0;
    transform: skew(32deg);
    width: 186px;
    z-index: 0; // So text is above bg color
  }

  &.navbar__item--active::after {
    border-bottom-style: none;
    z-index: 1;
  }
  
  // Adjust all items after the active item
  &.navbar__item--active ~ .navbar__item {
    left: -10px;
  }

  // Adjust item immediately following active item
  &.navbar__item--active + .navbar__item::before {
    height: 19px;
  }

  &.navbar__item--active + .navbar__item::after {
    border-left-style: none;
    left: 0;
    width: 170px;
  }
}

//// Theme -> Tabs -> Palette ////

// Orange

.navbar--theme-tabs.navbar--palette-orange {
  .navbar__item::after {
    background-color: #ffe37d;
  }

  .navbar__item.navbar__item--active::after {
    background-color: #fff;
  }
}

// Green

.navbar--theme-tabs.navbar--palette-green {
  .navbar__item::after {
    background-color: #c0db6d;
  }
  
  .navbar__item.navbar__item--active::after {
    background-color: #fff;
  }
}
