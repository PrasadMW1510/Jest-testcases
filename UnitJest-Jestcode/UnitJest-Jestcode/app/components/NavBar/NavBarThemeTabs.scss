// This file will be deleted once all navbars are updated to refactored version.
.nav-bar__theme-tabs .nav-bar__nav-item {
  align-items: center;
  border-style: none;
  box-sizing: border-box;
  display: inline-flex;
  height: 32px;
  overflow: hidden;
  position: relative;
  text-decoration: none;
  width: 170px;

  // Active and last item are always in full view
  &.nav-bar__nav-item--active,
  &:last-child {
    width: 180px;
  }

  &.nav-bar__nav-item--active:last-child {
    margin-right: 10px;
  }

  // Create Borders
  &::after {
    background: #000;
    content: '';
    height: 32px;
    left: 0;
    position: absolute;
    top: 0;
    width: 2px;
    z-index: 9;
  }

  &::before {
    border: 2px #000 solid;
    box-sizing: border-box;
    content: '';
    height: 32px;
    left: -16px;
    position: absolute;
    top: 0;
    transform: skew(32deg);
    width: 186px;
    z-index: 0; // So text is above bg color
  }

  &.nav-bar__nav-item--active::before {
    border-bottom-style: none;
  }

  &.nav-bar__nav-item--active ~ .nav-bar__nav-item {
    left: -10px;
  }

  &.nav-bar__nav-item--active + .nav-bar__nav-item::after {
    height: 19px;
  }

  &.nav-bar__nav-item--active + .nav-bar__nav-item::before {
    border-left-style: none;
    left: 0;
    width: 170px;
  }

  .nav-bar__text {
    color: #000;
    font-weight: 700;
    line-height: 26px; // Matches icon image height
    padding: 0 16px;
    position: relative;
  }
}
