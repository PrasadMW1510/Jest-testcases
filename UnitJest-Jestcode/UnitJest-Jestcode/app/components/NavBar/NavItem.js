import React from 'react';
import PropTypes from 'prop-types';

function NavItem({ active, children, className, component: Component, hasErrors, ...props }) {
  const classes = `navbar__item ${active ? 'navbar__item--active' : ''} ${className}`;
  return (
    <Component className={classes} {...props}>
      <span className={`navbar__item-text ${hasErrors ? 'navbar__item-text--error' : ''}`}>
        {children}
      </span>
    </Component>
  );
}

NavItem.defaultProps = {
  active: false,
  className: '',
  component: 'button',
  hasErrors: false,
};

NavItem.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  component: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  hasErrors: PropTypes.bool.isRequired,
};

export default NavItem;
