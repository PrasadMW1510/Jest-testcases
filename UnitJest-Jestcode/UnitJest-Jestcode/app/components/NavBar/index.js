/**
 *
 * NavBar
 *
 */

export { default } from './NavBar';
export NavItem from './NavItem';
