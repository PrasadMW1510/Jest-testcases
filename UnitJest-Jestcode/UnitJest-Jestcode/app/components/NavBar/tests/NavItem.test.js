import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavItem } from 'components/NavBar';

describe('<NavItem />', () => {
  it('Should render correctly', () => {
    const wrapper = shallow(
      <NavItem
        className="custom-class"
        style={{
          foo: 'test-if-custom-prop-gets-passed-through-to-dom-element',
        }}
      >
        navitem content
      </NavItem>
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Should render correctly with a custom component specified', () => {
    const wrapper = shallow(<NavItem component="a">navitem content</NavItem>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Should render correctly when flagged as active', () => {
    const wrapper = shallow(<NavItem active>navitem content</NavItem>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('Should render correctly when flagged with errors', () => {
    const wrapper = shallow(<NavItem hasErrors>navitem content</NavItem>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
