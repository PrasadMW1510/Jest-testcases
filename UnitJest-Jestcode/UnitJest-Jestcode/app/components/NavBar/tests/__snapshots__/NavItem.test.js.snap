// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> Should render correctly 1`] = `
<button
  className="navbar__item  custom-class"
  style={
    Object {
      "foo": "test-if-custom-prop-gets-passed-through-to-dom-element",
    }
  }
>
  <span
    className="navbar__item-text "
  >
    navitem content
  </span>
</button>
`;

exports[`<NavItem /> Should render correctly when flagged as active 1`] = `
<button
  className="navbar__item navbar__item--active "
>
  <span
    className="navbar__item-text "
  >
    navitem content
  </span>
</button>
`;

exports[`<NavItem /> Should render correctly when flagged with errors 1`] = `
<button
  className="navbar__item  "
>
  <span
    className="navbar__item-text navbar__item-text--error"
  >
    navitem content
  </span>
</button>
`;

exports[`<NavItem /> Should render correctly with a custom component specified 1`] = `
<a
  className="navbar__item  "
>
  <span
    className="navbar__item-text "
  >
    navitem content
  </span>
</a>
`;
